test/integration/dsn_test.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+
+from pyexasol import ExaConnectionError
+
+
+@pytest.mark.configuration
+def test_ip_range_with_custom_Port(connection):
+    dsn = "127.0.0.10..19:8564"
+    expected = {
+        ("127.0.0.10", "127.0.0.10", 8564, None),
+        ("127.0.0.14", "127.0.0.14", 8564, None),
+        ("127.0.0.13", "127.0.0.13", 8564, None),
+        ("127.0.0.18", "127.0.0.18", 8564, None),
+        ("127.0.0.17", "127.0.0.17", 8564, None),
+        ("127.0.0.16", "127.0.0.16", 8564, None),
+        ("127.0.0.15", "127.0.0.15", 8564, None),
+        ("127.0.0.12", "127.0.0.12", 8564, None),
+        ("127.0.0.19", "127.0.0.19", 8564, None),
+        ("127.0.0.11", "127.0.0.11", 8564, None),
+    }
+    actual = set(connection._process_dsn(dsn))
+    assert actual == expected
+
+
+@pytest.mark.configuration
+def test_multiple_ranges_with_multiple_ports_and_default_port_at_the_end(connection):
+    dsn = "127.0.0.10..19:8564,127.0.0.20,localhost:8565,127.0.0.21..23"
+    expected = {
+        ("127.0.0.13", "127.0.0.13", 8564, None),
+        ("localhost", "127.0.0.1", 8565, None),
+        ("127.0.0.16", "127.0.0.16", 8564, None),
+        ("127.0.0.20", "127.0.0.20", 8565, None),
+        ("127.0.0.22", "127.0.0.22", 8563, None),
+        ("127.0.0.19", "127.0.0.19", 8564, None),
+        ("127.0.0.18", "127.0.0.18", 8564, None),
+        ("127.0.0.10", "127.0.0.10", 8564, None),
+        ("127.0.0.15", "127.0.0.15", 8564, None),
+        ("127.0.0.17", "127.0.0.17", 8564, None),
+        ("127.0.0.11", "127.0.0.11", 8564, None),
+        ("127.0.0.23", "127.0.0.23", 8563, None),
+        ("127.0.0.12", "127.0.0.12", 8564, None),
+        ("127.0.0.14", "127.0.0.14", 8564, None),
+        ("127.0.0.21", "127.0.0.21", 8563, None),
```

Comment:
Why is the expected data unordered? Makes reading much more difficult.....

Reply:
Actually this is the order which was initially returned, anyway I would not want to make assumptions what the used order will be as long as it is parsed correctly.